hall-of-fame-service/src/main/java/hcmus/alumni/halloffame/controller/HallOfFameServiceController.java

```diff
@@ -0,0 +1,243 @@
+				if (thumbnail.getSize() > 5 * 1024 * 1024) {
+					return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("File must be lower than 5MB");
+				}
+
+				// Save new thumbnail
+				String thumbnailUrl = imageUtils.saveImageToStorage(imageUtils.getHofPath(id), thumbnail, "thumbnail");
+				halloffame.setThumbnail(thumbnailUrl);
+			}
+			if (!faculty.equals("")) {
+				halloffame.setFaculty(new FacultyModel(faculty));
+			}
+			if (beginningYear != null) {
+				halloffame.setBeginningYear(beginningYear);
+			}
+
+			if (!emailOfUser.equals("")) {
+				String userId = halloffameRepository.getUserIdByEmail(emailOfUser);
+				halloffame.setUserId(new UserModel(userId));
+			}
+
+			// Save hall of fame
+			halloffameRepository.save(halloffame);
+
+		} catch (IOException e) {
+			System.err.println(e);
+			return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Failed to update hall of fame");
+		}
+
+		return ResponseEntity.status(HttpStatus.OK).body("Updated successfully!");
+	}
+
+	@PreAuthorize("hasAnyAuthority('Admin')")
+	@PutMapping("/{id}/content")
+	public ResponseEntity<String> updateHallOfFameContent(@PathVariable String id,
+			@RequestBody(required = false) HallOfFameModel updatedHallOfFame) {
+		if (updatedHallOfFame.getContent() == null) {
+			return ResponseEntity.status(HttpStatus.OK).body("");
+		}
+		// Find hof
+		Optional<HallOfFameModel> optionalHallOfFame = halloffameRepository.findById(id);
+		if (optionalHallOfFame.isEmpty()) {
+			return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Invalid id");
+		}
+		HallOfFameModel halloffame = optionalHallOfFame.get();
+		if (!updatedHallOfFame.getContent().equals("")) {
+			String newContent = imageUtils.updateImgFromHtmlToStorage(halloffame.getContent(), updatedHallOfFame.getContent(), id);
+			if (newContent.equals(halloffame.getContent())) {
+				return ResponseEntity.status(HttpStatus.OK).body("");
+			}
+			halloffame.setContent(newContent);
+			halloffameRepository.save(halloffame);
+		}
+		return ResponseEntity.status(HttpStatus.OK).body("");
+	}
+	
+	@PreAuthorize("hasAnyAuthority('Admin')")
+	@DeleteMapping("/{id}")
+	public ResponseEntity<String> deleteHallOfFame(@PathVariable String id) {
+	    Optional<HallOfFameModel> optionalHallOfFame = halloffameRepository.findById(id);
+	    if (optionalHallOfFame.isEmpty()) {
+	        return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Invalid id");
+	    }
+	    HallOfFameModel hof = optionalHallOfFame.get();
+	    hof.setStatus(new StatusPostModel(4));
+	    halloffameRepository.save(hof);
+	    return ResponseEntity.status(HttpStatus.OK).body("Updated status successfully!");
+	}
+	
+	@GetMapping("/{id}")
+	public ResponseEntity<HallOfFameModel> getHallOfFameById(@PathVariable String id) {
+	    Optional<HallOfFameModel> halloffame = halloffameRepository.findById(id);
+	    return halloffame.map(ResponseEntity::ok)
+	        .orElseGet(() -> ResponseEntity.status(HttpStatus.NOT_FOUND).build());
+	}
+}
```

Comment:
```suggestion
	public ResponseEntity<IHallOfFameDto> getHallOfFameById(@PathVariable String id) {
	    Optional<IHallOfFameDto> halloffame = halloffameRepository.findById(id);
	    return halloffame.map(ResponseEntity::ok)
	        .orElseGet(() -> ResponseEntity.status(HttpStatus.NOT_FOUND).build());
	}
}
```
dùng hàm viewsIncrement để tăng views nếu tìm thấy